activefires_pp/tests/test_spatiotemporal_alarm_filtering.py

```diff
@@ -328,15 +334,35 @@ def test_alarm_filter_runner_init(setup_comm, get_config):
                                     'restapi_url': 'https://xxx.smhi.se:xxxx'}
 
 
+@patch('activefires_pp.spatiotemporal_alarm_filtering.get_xauth_environment_variable')
+@patch('activefires_pp.spatiotemporal_alarm_filtering.read_config')
+@patch('activefires_pp.spatiotemporal_alarm_filtering.AlarmFilterRunner._setup_and_start_communication')
+def test_alarm_filter_runner_init_no_env(setup_comm, get_config, get_xauth):
+    """Test initialize the AlarmFilterRunner class."""
+    get_config.return_value = CONFIG_EXAMPLE
+    get_xauth.return_value = None
+
+    myconfigfile = "/my/config/file/path"
+
+    with pytest.raises(OSError) as exec_info:
+        _ = AlarmFilterRunner(myconfigfile)
+
+    expected = "Environment variable XAUTH_SMHI_FIREALARMS_REST_API not set!"
+    assert str(exec_info.value) == expected
+
+
+@patch('activefires_pp.spatiotemporal_alarm_filtering.get_xauth_environment_variable')
```

Comment:
That's a lot of patching! Wouldn't a patched listener and publisher suffice?

Reply:
I am using more fixtures now, so this should be better now I hope.